### DISCUSSION TITLE
Is Klogg real-time?

### DISCUSSION THREAD

**Discussion by heltoncargnin** [variar/klogg#739] [Q&A]
Hello everyone, how are you?

I installed klogg to analyze the logs of one of my applications.

However, I noticed that it is not in real time. Is there a parameter to make the tail in real time?

Best regards,
Helton

**Comment by variar** [comment]
Hi, there is a button to enable "Follow" mode: 
![image](https://github.com/user-attachments/assets/765b0cf2-b873-487d-8cca-8d68b9697408)

Also there is a shortcut. And by default when you try to scroll past the end of file klogg should enable "follow" mode.

**Comment by heltoncargnin** [comment]
Good morning,
 Thank you for your reply.

 
 
 I noticed that even with the "follow" option enabled, it is not in real time.

You can see the screenshots below:

Kloog screen:

Last time 2024-11-19 07:52:46,572

![image](https://github.com/user-attachments/assets/a9bcfa27-0805-4ea2-852e-9d9206de72c3)



Log screen in Linux:

2024-11-19 07:54:06,146

![image](https://github.com/user-attachments/assets/7f857faf-7abd-4524-a94a-0e45f9bd97f0)


Therefore, it does not update in real time.

I tested it directly with notepad++ and also with baretail, and it is in real time.

**Reply by variar** [reply]
Sometimes OS notifications about file changes are not delivered properly. For such cases you can enable polling for changes in klogg settings. And might worth trying enabling/disabling fast modification detection.

**Comment by heltoncargnin** [comment]
Hello!

Thanks for the tip!

However, it didn't work, it still doesn't update in real time.

Here's a screenshot of how it is:

![image](https://github.com/user-attachments/assets/46f062a7-377a-47ed-a98b-17cc4bf4ceef)

**Reply by variar** [reply]
Polling for changes should work. What filesystem is being used? What is the update pattern of the file? Is it just new lines being added at the end of the file?

Is the view in follow file mode? It should have a visible yellow pattern at the bottom of the view and scroll shoud be at the end of file.  Are both views not updating or only the filtered one?

**Comment by heltoncargnin** [comment]
Answering your questions:
The log file is on a Linux server and shared via Samba for the user to check on Windows
The update standard is in real time, in other applications such as Notepad++ or Baretail, it updates in real time.

Yes, the yellow bar appears and is at the end of the scroll bar. Both views are not appearing in real time.

![image](https://github.com/user-attachments/assets/1856c2e3-31e9-4b67-b006-0444086b376c)

**Comment by heltoncargnin** [comment]
can you help me

**Comment by heltoncargnin** [comment]
I need to solve this case, can you help me?
